spec/controllers/inbox_controller_spec.rb

```diff
@@ -0,0 +1,192 @@
+require "spec_helper"
+
+describe InboxController do
+  include LoginMacros
+  include RedirectExpectationHelper
+  let(:user) { create(:user) }
+
+  describe "GET #show" do
+    it "redirects to user page when not logged in" do
+      get :show, user_id: user.login
+      it_redirects_to user_path(user)
+    end
+
+    it "redirects to user page when logged in as another user" do
+      fake_login_known_user(create(:user))
+      get :show, user_id: user.login
+      it_redirects_to user_path(user)
+    end
+
+    context "when logged in as the same user" do
+      before { fake_login_known_user(user) }
+
+      it "renders the user inbox" do
+        get :show, user_id: user.login
+        expect(response).to render_template("show")
+        expect(assigns(:inbox_total)).to eq(0)
+        expect(assigns(:unread)).to eq(0)
+      end
+
+      context "with unread comments" do
+        let!(:inbox_comments) do
+          Array.new(3) do |i|
+            create(:inbox_comment, user: user, created_at: Time.now + i.days)
```

Comment:
If I don't space them out, they can get created in the same second and break the ordering test. This is a low-budget "timecop".